opensearch-operator/pkg/builders/cluster_test.go

```diff
@@ -91,6 +91,30 @@ var _ = Describe("Builders", func() {
 				Value: "master",
 			}))
 		})
+		It("should have annotations added to node", func() {
+			var clusterObject = ClusterDescWithVersion("1.3.0")
+			var nodePool = opsterv1.NodePool{
+				Component: "masters",
+				Roles:     []string{"cluster_manager"},
+				Annotations: map[string]string{
+					"testAnnotationKey": "testAnnotationValue",
+				},
+			}
+			var result = NewSTSForNodePool("foobar", &clusterObject, nodePool, "foobar", nil, nil, nil)
+			Expect(result.Spec.Template.Annotations).To(Equal(map[string]string{
+				"testAnnotationKey": "testAnnotationValue",
```

Comment:
An annotation `"opster.io/config":  "foobar"` is also added by the operator code. You either need to add this for the comparison as well or only check if the one key/value exists.

Reply:
Add the check for the generated annotation as requested.